### PR TITLE
Split Tizen.NUI (#848)

### PR DESCRIPTION
**Pull request by AdunFang**
### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI.Xaml/src/public/Xaml/XamlServiceProvider.cs [hunk 1]
```diff
@@ -2,11 +2,16 @@
...
 using Tizen.NUI.Binding.Internals;
 
 namespace Tizen.NUI.Xaml
 {
-    internal class XamlServiceProvider : IServiceProvider
+    /// <summary>
+    /// The class to provide xaml service.
+    /// </summary>
+    /// <since_tizen> 5 </since_tizen>
+    public class XamlServiceProvider : IServiceProvider
```
Comment (elishateng) [inline src/Tizen.NUI.Xaml/src/public/Xaml/XamlServiceProvider.cs]: Should we add [EditorBrowsable(EditorBrowsableState.Never)] in public class ?

File: src/Tizen.NUI.Xaml/src/public/Xaml/XamlServiceProvider.cs [hunk 2]
```diff
@@ -132,16 +149,28 @@ IEnumerable<object> IProvideParentValues.ParentObjects
         }
     }
 
-    internal class SimpleValueTargetProvider : IProvideParentValues, IProvideValueTarget
+    /// <summary>
+    /// The class to provide simple value target.
+    /// </summary>
+    /// <since_tizen> 5 </since_tizen>
+    public class SimpleValueTargetProvider : IProvideParentValues, IProvideValueTarget, IReferenceProvider
```
Comment (elishateng) [inline src/Tizen.NUI.Xaml/src/public/Xaml/XamlServiceProvider.cs]: Should we add [EditorBrowsable(EditorBrowsableState.Never)] in public class ?

File: src/Tizen.NUI.Xaml/src/public/Xaml/XamlServiceProvider.cs [hunk 3]
```diff
@@ -167,14 +196,41 @@ object IProvideValueTarget.TargetProperty
...
+            return null;
+        }
     }
 
-    internal class XamlTypeResolver : IXamlTypeResolver
+    /// <summary>
+    /// The class to resolve xaml type.
+    /// </summary>
+    /// <since_tizen> 5 </since_tizen>
+    public class XamlTypeResolver : IXamlTypeResolver
```
Comment (elishateng) [inline src/Tizen.NUI.Xaml/src/public/Xaml/XamlServiceProvider.cs]: Should we add [EditorBrowsable(EditorBrowsableState.Never)] in public class ?

File: src/Tizen.NUI.Xaml/src/public/Xaml/XmlLineInfo.cs [hunk 4]
```diff
@@ -2,7 +2,11 @@
 
 namespace Tizen.NUI.Xaml
 {
-    internal class XmlLineInfo : IXmlLineInfo
+    /// <summary>
+    /// The class XmlLineInfo.
+    /// </summary>
+    /// <since_tizen> 5 </since_tizen>
+    public class XmlLineInfo : IXmlLineInfo
```
Comment (elishateng) [inline src/Tizen.NUI.Xaml/src/public/Xaml/XmlLineInfo.cs]: Should we add [EditorBrowsable(EditorBrowsableState.Never)] in public class ?

File: src/Tizen.NUI.Xaml/src/internal/XamlBinding/DataTemplateSelector.cs [hunk 5]
```diff
@@ -9,17 +9,30 @@ internal abstract class DataTemplateSelector : DataTemplate
 
         public DataTemplate SelectTemplate(object item, BindableObject container)
         {
+            // var listView = container as ListView;
```
Comment (m-macias) [inline src/Tizen.NUI.Xaml/src/internal/XamlBinding/DataTemplateSelector.cs]: Are those commented lines necessary?

File: src/Tizen.NUI.Xaml/src/internal/XamlBinding/CollectionSynchronizationCallback.cs [hunk 6]
```diff
@@ -0,0 +1,7 @@
+using System;
+using System.Collections;
+
+namespace Tizen.NUI.Binding
+{
+    public delegate void CollectionSynchronizationCallback(IEnumerable collection, object context, Action accessMethod, bool writeAccess);
```
Comment (m-macias) [inline src/Tizen.NUI.Xaml/src/internal/XamlBinding/CollectionSynchronizationCallback.cs]: New public delegate added.

File: src/Tizen.NUI.Xaml/src/public/Forms/BaseComponents/ImageView.cs [hunk 7]
```diff
@@ -0,0 +1,477 @@
...
+using Tizen.NUI;
+using static Tizen.NUI.BaseComponents.ImageView;
+
+namespace Tizen.NUI.Xaml.Forms.BaseComponents
+{
+    /// <summary>
+    /// ImageView is a class for displaying an image resource.<br />
+    /// An instance of ImageView can be created using a URL or an image instance.<br />
+    /// </summary>
+    /// <since_tizen> 5.5 </since_tizen>
```
Comment (m-macias) [inline src/Tizen.NUI.Xaml/src/public/Forms/BaseComponents/ImageView.cs]: Should be `<since_tizen>6</since_tizen>`

File: src/Tizen.NUI/src/public/Property.cs [hunk 8]
```diff
@@ -182,18 +181,6 @@ public void PushBack(PropertyValue value)
             if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
         }
 
-        /// <summary>
-        /// Adds an keyvalue to the array.
-        /// This function should be first
-        /// </summary>
-        /// <param name="value">The value to add at the end of the array.</param>
-        public PropertyArray Add(KeyValue value)
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Property.cs]: Public API removed without Deprecation.

File: src/Tizen.NUI/src/public/BaseHandle.cs [hunk 9]
```diff
@@ -464,6 +439,52 @@ internal RefObject GetObjectPtr()
...
+        internal void SetValue(BindableProperty property, object value)
+        {
+            property.PropertyChanged?.Invoke(this, null, value);
+        }
+
+        /// <summary>
+        /// The event args of dispose.
+        /// </summary>
+        /// <since_tizen> 5.5 </since_tizen>
+        public class DisposeEventArgs : EventArgs
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/BaseHandle.cs]: Public API added.

File: src/Tizen.NUI/src/public/Animation.cs [hunk 10]
```diff
@@ -425,66 +417,6 @@ public float ProgressNotification
             }
         }
 
-        /// <summary>
-        /// Gets or sets the properties of the animation.
-        /// </summary>
-        public string[] Properties
```
Comment (m-macias) [inline src/Tizen.NUI/src/public/Animation.cs]: Public API removed without Deprecation.

File: src/Tizen.NUI/src/internal/Application.cs [hunk 11]
```diff
@@ -309,445 +307,18 @@ public IntPtr VoidP
...
-    /// <summary>
-    /// A class to get resources in current application.
-    /// </summary>
-    public class GetResourcesProvider
+    internal class Application : BaseHandle
     {
-        /// <summary>
-        /// Get resources in current application.
-        /// </summary>
-        static public IResourcesProvider Get()
```
Comment (m-macias) [inline src/Tizen.NUI/src/internal/Application.cs]: Public API removed without Deprecation.

**Pull request merged**